BSP/Simulator/Hardware/Contactor.py

```diff
@@ -14,15 +17,19 @@ def read():
     """
     os.makedirs(os.path.dirname(file), exist_ok=True)
     if not os.path.exists(file):
-        with open(file, 'w'):
+        with open(file, "w"):
             pass
 
-    with open(file, 'r') as csvfile:
-        fcntl.flock(csvfile.fileno(), fcntl.LOCK_EX)    # Lock file before reading
-        try:    #the following code is to prevent simulator from crashing if the csv is blank
-            csvreader = csv.reader(csvfile)
-            contactor_states = next(csvreader)
-        except StopIteration as stopIt:
-            contactor_states = ['', '']
-        fcntl.flock(csvfile.fileno(), fcntl.LOCK_UN)    # Unlock file after reading
-        return contactor_states 
+    states = []
+    with open(file, "r") as csvfile:
+        fcntl.flock(csvfile.fileno(), fcntl.LOCK_EX)  # Lock file before reading
+        csvreader = csv.reader(csvfile)
+        for row in csvreader:
+            states.append(row)
+        fcntl.flock(csvfile.fileno(), fcntl.LOCK_UN)
+    try:
+        states = int(states[2][0])
```

Comment:
So the state of both contactors is going to be just glued to one another as a single integer (which we later separate and return individually by left shifting), right? Just to clarify for my own understanding

Reply:
They're connected through GPIO, so both contactors are pins on one port. In that sense, their states are accessed/modified with one single integer where each contactor is represented by an individual bit.